source/Condition.h

```diff
@@ -0,0 +1,194 @@
+	// If it was a condition, the key must be the same (value doesn't matter)
+	// If it was a literal (no key) then the value must be the same.
+	// If one is literal and the other is conditional, the result is false.
+	bool SameOrigin(const Condition<V,K> &o);
+
+	// Does this originate from a condition?
+	bool HasConditions() const { return !key.empty(); }
+
+	// Does this originate from a literal value (ie. 5.071)?
+	bool IsLiteral() const { return key.empty(); }
+
+	// Floating-point values are false if they're within half the
+	// type's precision of 0 while any other types are passed
+	// through static_cast<bool>
+	explicit operator bool() const;
+
+	// Allow the Condition to be treated as its value in most contexts.
+	operator ValueType() const { return value; }
+
+private:
+	ValueType value;
+	KeyType key;
+};
+
+
+// Allow construction and assignment between Condition types to
+// facilitate type conversion.
+template <class V, class K>
+template <class V2, class K2>
+Condition<V,K>::Condition(const Condition<V2,K2> &other):
+	value(static_cast<ValueType>(other.Value())),
+	key(static_cast<KeyType>(other.Key()))
+{
+}
+
+
+// Allow construction and assignment between Condition types to
+// facilitate type conversion.
+template <class V, class K>
+template <class V2, class K2>
+Condition<V,K> &Condition<V,K>::operator = (const Condition<V2,K2> &other)
+{
+	value = static_cast<ValueType>(other.Value());
+	key = static_cast<KeyType>(other.Key());
+	return *this;
+}
+
+
+// Update the value from a scope that contains it
+template <class V, class K>
+template <class Getter>
+const V &Condition<V,K>::UpdateConditions(const Getter &getter)
+{
+	// If this was a literal, there is nothing to update.
+	if(HasConditions())
+	{
+		auto got = getter.HasGet(key);
+		// Assumes: got.first = true iff getter has key
+		// got.second = value iff got.first
+		if(got.first)
+			value = static_cast<ValueType>(got.second);
+	}
+	return value;
+}
+
+
+// Update the value from a scope that contains it, if the new value passes a validator.
+template <class V, class K>
+template <class Getter, class Validator>
+const V &Condition<V,K>::UpdateConditions(const Getter &getter, Validator validator)
+{
+	// If this was a literal, there is nothing to update.
+	if(HasConditions())
+	{
+		auto got = getter.HasGet(key);
+		// Assumes: got.first = true iff getter has key
+		// got.second = value iff got.first
+		if(got.first && validator(got.second))
+		{
+			value = static_cast<ValueType>(got.second);
+			return value;
+		}
+	}
+	if(!validator(value))
+		value = ValueType();
+	return value;
+}
+
+
+template <class V, class K>
+bool Condition<V,K>::SameOrigin(const Condition<V,K> &o)
+{
+	if(HasConditions())
+		return key == o.Key();
+	else if(o.HasConditions())
+		return false;
+	else
+		return value == o.Value();
+}
+
+
+template < class T, typename std::enable_if<std::is_floating_point<T>::value, T>::type* Check = nullptr >
+bool NotNearZero(T whut)
```

Comment:
whut?
I guess that isn't intended to stay the name right?

Reply:
Changed to "number."